core/src/main/java/org/apache/accumulo/core/data/Key.java

```diff
@@ -185,6 +185,44 @@ public Key(byte row[], int rOff, int rLen, byte cf[], int cfOff, int cfLen, byte
   }
 
   /**
+   * Creates a key.
+   *
+   * @param row
+   *          bytes containing row ID
+   * @param rOff
+   *          offset into row where key's row ID begins (inclusive)
+   * @param rLen
+   *          length of row ID in row
+   * @param cf
+   *          bytes containing column family
+   * @param cfOff
+   *          offset into cf where key's column family begins (inclusive)
+   * @param cfLen
+   *          length of column family in cf
+   * @param cq
+   *          bytes containing column qualifier
+   * @param cqOff
+   *          offset into cq where key's column qualifier begins (inclusive)
+   * @param cqLen
+   *          length of column qualifier in cq
+   * @param cv
+   *          bytes containing column visibility
+   * @param cvOff
+   *          offset into cv where key's column visibility begins (inclusive)
+   * @param cvLen
+   *          length of column visibility in cv
+   * @param ts
+   *          timestamp
+   * @param deleted
+   *          delete marker
+   * @param copy
+   *          if true, forces copy of byte array values into key
+   */
+  public Key(byte row[], int rOff, int rLen, byte cf[], int cfOff, int cfLen, byte cq[], int cqOff, int cqLen, byte cv[], int cvOff, int cvLen, long ts, boolean deleted, boolean copy) {
```

Comment:
What makes this constructor different from the others and not worthy to be exposed to the client? It's a low level constructor that sets all the possible components of `Key` in the proper way. There is another public constructor that is basically this but with `deleted` defaulting to `false` and `copy` to `true` and I don't see why changing those two flags could have any undesired effects.